src/lib/list/list-option.html

```diff
@@ -0,0 +1,11 @@
+<div [ngClass]="(checkboxPosition == 'before')?'mat-list-item-content' : 'mat-list-item-content-reverse'" >
```

Comment:
Instead of using `ngClass` and applying one of two classes, you should be able to have the default styles always applied and then apply the "reverse" styles when the condition is met  like this:
```html
<div [class.mat-list-option-reverse]="checkboxPosition == 'after'">
```